lib/rest_client/core_ext/logged_request.rb

```diff
@@ -18,6 +18,7 @@ def logged_request(opts = {}, &block)
     time = (Time.now - started).round(2)
     content_type = content_type_from_headers(opts[:headers])
     payload = filter(content_type).new(data: opts[:payload]).filter
+    opts[:headers].except!(:authorization) if opts[:headers]
```

Comment:
We should also be cautious of method calls that supply the `Authorization` header as a string - not a symbol. 

i.e.

```ruby
RestClient::Request.logged_request(method: :get, url: 'https://example.com/', headers: { 'Authorization' => 'basicauthpass' })
```

What if we do something like this?

```ruby
  def filter_request_params(params)
    headers = params.fetch(:headers, {})
    filtered = headers.reject { |k, _| k.to_s.casecmp('authorization').zero? }
    params.merge(headers: filtered)
  end

  def request_data(params)
    {
      request: filter_request_params(params)
    }
  end
```